src/strongbus_spec.ts

```diff
@@ -2238,4 +2238,66 @@ describe('Strongbus.Bus', () => {
       });
     });
   });
-});
\ No newline at end of file
+
+  const {gc} = global;
+  if(gc) {
+    describe('it does not leak memory', () => {
+      const numCalls = 10000;
+
+      function generateId(): string {
+        return (
+          Math.random().toString(36).substring(2, 15) +
+          Math.random().toString(36).substring(2, 15)
+        );
+      }
+
+      function sample<T>(array: T[]): T {
+        const idx = Math.floor(array.length * Math.random());
+        return array[idx];
+      }
+
+      function bytes(bytes: number): string {
+        const mb = Math.pow(2, 20);
+        return `${(bytes / mb).toFixed(2)}mb`
+      }
+
+      it('Strongbus.Bus does not leak memory', async () => {
+        const bus = new Strongbus.Bus<any>();
+        const sub = bus.on('*', () => {});
+        const events = new Array(100).fill(0).map(generateId);
+
+        // Take baseline memory snapshot
+        gc();
+        const memStart = process.memoryUsage().heapUsed;
+
+        // Emit a bunch of events
+        // In theory, this shouldn't increase heap size
+        for(let i = 0; i < numCalls; i++) {
+          const event = sample(events);
+          bus.emit(event, generateId());
+        }
+
+        // Wait for msgbus promises to resolve
+        await new Promise((resolve) => {
+          setTimeout(resolve, 0);
+        });
+
+        // Take final memory snapshot
+        gc();
+        const memEnd = process.memoryUsage().heapUsed;
+        if(memEnd - memStart > 0) {
```

Comment:
I think throwing here rather than console erroring and executing an expect statement would be the succinct approach